### DISCUSSION TITLE
edit source code for custom keybindings.

### DISCUSSION THREAD

**Discussion by minsoehan** [Cloudef/bemenu#396] [Q&A] [answered]
I am checking source code of bemenu to edit for custom keybindings. I want bracketleft to go down and bracketright to go up. I found _lib/renderers/wayland/wayland.c_ file and edited as follows:

    switch (sym) {
        case XKB_KEY_Up:
        case XKB_KEY_bracketright:
            return BM_KEY_UP;
       
        case XKB_KEY_Down:
        case XKB_KEY_bracketleft:
            return BM_KEY_DOWN;
    ...

Then built it by running
`$ make clients wayland`
It does not work. What I am missing or is it possible?

**Comment by Cloudef** [comment]
After rebuilding are you making sure the new renderer is being used?
` BEMENU_RENDERERS=. `  to set path to CWD

**Reply by minsoehan** [reply]
Oh! I didn't. let me try.

**Reply by Cloudef** [reply, marked as answer]
For running source build completely use:
```sh
# To test from source, you have to point the LD_LIBRARY_PATH and BEMENU_RENDERERS variables:
LD_LIBRARY_PATH=. BEMENU_RENDERERS=. ./bemenu-run
```
As mentioned in readme, or simply `nix run .` if using nix (compiles and runs bemenu-run)

**Reply by minsoehan** [reply]
Yes, Thanks, it works now.
